hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MetricsUtil.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.metrics.impl;
+
+import com.hazelcast.internal.metrics.MetricTarget;
+import com.hazelcast.internal.metrics.ProbeLevel;
+
+import java.util.Collection;
+import java.util.Set;
+
+import static com.hazelcast.internal.metrics.MetricTarget.ALL_TARGETS;
+import static com.hazelcast.internal.metrics.MetricTarget.ALL_TARGETS_BUT_DIAGNOSTICS;
+import static com.hazelcast.internal.metrics.MetricTarget.DIAGNOSTICS;
+import static com.hazelcast.internal.metrics.MetricTarget.asSet;
+import static com.hazelcast.internal.metrics.ProbeLevel.DEBUG;
+import static java.util.Collections.emptySet;
+
+final class MetricsUtil {
+    private MetricsUtil() {
+    }
+
+    static Collection<MetricTarget> excludedTargets(Collection<MetricTarget> exclusions, ProbeLevel level) {
+        if (DEBUG != level) {
+            return exclusions;
+        } else if (exclusions.contains(DIAGNOSTICS)) {
+            return ALL_TARGETS;
```

Comment:
Does this branch mean that probes that include `DIAGNOSTICS` in their exclusion list won't be collected at all when `DEBUG` level is enabled?

Reply:
Well, it's collected but at least no out-of-the-box publisher will pick it up. You can register a custom publisher programmatically with internal API though where you can do anything.

I don't know though if configuring a metric `DEBUG` but with `DIAGNOSTICS` excluded actually makes any sense. `DEBUG` metrics are for diagnostics. But we need to cover this possible case and make sure we don't fail.